frontend/packages/dev-console/src/components/pipelines/modals/triggers/TriggerBindingSelector.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import { useFormikContext } from 'formik';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { Firehose } from '@console/internal/components/utils';
+import { TriggerBindingModel } from '../../../../models';
+import { TriggerBindingKind } from '../../resource-types';
+import TriggerBindingSelectorDropdown from './TriggerBindingSelectorDropdown';
+import { AddTriggerFormValues } from './types';
+
+type TriggerBindingSelectorProps = {
+  description?: string;
+  label?: string;
+  onChange: (selectedTriggerBinding: TriggerBindingKind) => void;
+};
+
+const TriggerBindingSelector: React.FC<TriggerBindingSelectorProps> = (props) => {
+  const { description, label = TriggerBindingModel.label, onChange } = props;
+  const { values } = useFormikContext<AddTriggerFormValues>();
+
+  return (
+    <Firehose
```

Comment:
Why not use the `useK8sWatchResource` hook here?

Reply:
Good question. Likely because I wrote it before I was aware of it. This PR has been a long time coming haha.

Reply:
https://issues.redhat.com/browse/ODC-3610